src/main.rs

```diff
@@ -186,6 +191,11 @@ fn main() {
             };
         }
 
+        if let Some(id) = matches.value_of("ID") {
+            let id: u32 = id.parse::<u32>().expect("The id has to be an unsigned integer");
```

Comment:
the extra `: u32` is not really necessary here

Reply:
sure, the explicit call to `u32` really isn't necessary here